apps/io-wallet-user-func/src/infra/trial-system/client.ts

```diff
@@ -0,0 +1,66 @@
+import { TrialSystemApiClientConfig } from "@/app/config";
+import { UserTrialSubscriptionRepository } from "@/user";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import * as TE from "fp-ts/lib/TaskEither";
+
+import { TrialSystemHealthCheck } from "./health-check";
+
+export class TrialSystemClient
+  implements UserTrialSubscriptionRepository, TrialSystemHealthCheck
+{
+  #apiKey: string;
+  #baseURL: string;
+  #trialId: string;
+  featureFlag: string;
+
+  getUserSubscriptionDetail = (userId: NonEmptyString) =>
+    TE.tryCatch(
+      async () => {
+        const result = await fetch(
+          new URL(
+            `/trials/${this.#trialId}/subscriptions/${userId}`,
+            this.#baseURL,
+          ),
+          {
+            headers: {
+              "x-functions-key": this.#apiKey,
+            },
+            method: "GET",
+            signal: AbortSignal.timeout(3000),
+          },
+        );
+        if (!result.ok) {
+          throw new Error(JSON.stringify(await result.json()));
+        }
+        return result.json();
+      },
```

Comment:
Not on this PR, but what do you think about using the service's swagger and using the dynamically generated client?

Reply:
we can do it but it would be for just one endpoint